src/idex/common.js

```diff
@@ -0,0 +1,178 @@
+import { base64UrlEncode } from '../utils/b64'
+import { fromError } from '../utils/emitter'
+import { toParams } from '../utils/url'
+import { asParamOrEmpty, asStringParamWhen, asStringParam, expiresInHours, mapAsParams } from '../utils/types'
+import { DEFAULT_IDEX_AJAX_TIMEOUT, DEFAULT_IDEX_URL, DEFAULT_REQUESTED_ATTRIBUTES } from '../utils/consts'
+
+/**
+ * @typedef {Object} Cache
+ * @property {function} [get]
+ * @property {function} [set]
+ */
+
+/**
+ * @return {Cache}
+ */
+export function storageHandlerBackedCache (expirationHours, domain, storageHandler) {
+  const IDEX_STORAGE_KEY = '__li_idex_cache'
+
+  function _cacheKey (rawKey) {
+    if (rawKey) {
+      const suffix = base64UrlEncode(JSON.stringify(rawKey))
+      return `${IDEX_STORAGE_KEY}_${suffix}`
+    } else {
+      return IDEX_STORAGE_KEY
+    }
+  }
+
+  return {
+    get: (key) => {
+      const cachedValue = storageHandler.get(_cacheKey(key))
+      if (cachedValue) {
+        return JSON.parse(cachedValue)
+      } else {
+        return cachedValue
+      }
+    },
+    set: (key, value) => {
+      try {
+        storageHandler.set(
+          _cacheKey(key),
+          JSON.stringify(value),
+          expiresInHours(expirationHours),
+          domain
+        )
+      } catch (ex) {
+        fromError('IdentityResolverStorage', ex)
+      }
+    }
+  }
+}
+
+/**
+ * @type {Cache}
+ */
+export const noopCache = {
+  get: (key) => {
+    return null
+  },
+  set: (key, value) => {
+
+  }
```

Comment:
I would do it this way, but as you prefer
```suggestion
  get: (key) => null,
  set: (key, value) => undefined
```